api-client-specifications/recommend_api.md

```diff
@@ -0,0 +1,84 @@
+# Recommend API clients specifications
+
+## Table of Contents
+
+- [`recommend_client` interface](#recommend_client-interface)
+- [Objects](#objects)
+- [Responses](#responses)
+
+## `recommend_client` interface
+
+```java
+interface recommend_client {
+    function get_recommendations(queries: [recommendations_query]) return get_recommendations_response
+    function get_related_products(queries: [related_products_query]) return get_recommendations_response
```

Comment:
Not that this is in the scope of the PR, but does anyone know why we went with `getRelatedProducts` instead of `getRelatedObjects`? I don't know what you think, but to me it seems that with this name we disqualify any use case that's not e-commerce since they'll think this method is not for them.

Reply:
I agree, and we've used `objects` in the frontend in a few demos IIRC. I assume it's because it's a common naming in e-commerce, so it speaks to the user directly.

Reply:
This is the name of the Algolia Recommend feature: Frequently Bought Together and Related Products.

Reply:
It follows the "Related Products" AI Model fine-tuned for e-commerce.